client_test.go

```diff
@@ -375,25 +375,32 @@ func TestClientRefreshBehaviour(t *testing.T) {
 }
 
 func TestClientResurrectDeadSeeds(t *testing.T) {
-	seed1 := newMockBroker(t, 1)
-	seed2 := newMockBroker(t, 2)
-	seed3 := newMockBroker(t, 3)
-	addr1 := seed1.Addr()
-	addr2 := seed2.Addr()
-	addr3 := seed3.Addr()
-
+	initialSeed := newMockBroker(t, 0)
 	emptyMetadata := new(MetadataResponse)
-	seed1.Returns(emptyMetadata)
```

Comment:
can't you just create a goroutine that waits until `NewClient` is blocked, then reads off the seed order from the client and returns metadata to the first broker on that list?